skbio/stats/composition.py

```diff
@@ -807,6 +807,319 @@ def centralize(mat):
+
+@experimental(as_of="0.5.7")
+def _robust_pairwise_vlr(mat:np.array, ddof:int):
+    r"""
+    Performs pairwise variance log ratio transformation while masking zeros
+    
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+       
+    ids: array_like, str
+        component names
+        
+    ddof: int
+        degrees of freedom
+       
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+    # Mask zeros
+    X = np.ma.masked_array(mat, mask=mat == 0)
+    
+    # Log Transform
+    X_log = np.ma.log(X)
+    
+    # Variance Log Ratio
+    covariance = np.ma.cov(X_log.T, ddof=ddof) 
+    diagonal = np.ma.diagonal(covariance)
+    vlr_data = -2*covariance + diagonal[:,np.newaxis] + diagonal 
+    return vlr_data
+
+@experimental(as_of="0.5.7")
+def pairwise_vlr(mat, ids=None, ddof=1, robust=False) -> DistanceMatrix:
+    r"""
+    Performs pairwise variance log ratio transformation
+    
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+       
+    ids: array_like, str
+        component names
+        
+    ddof: int
+        degrees of freedom
+        
+    robust: bool
+        mask zeros at the cost of performance
+       
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+         
+    Examples
+    --------
+    >>> mat = np.asarray([
+        [1,2,3],
+        [5,8,13],
+        [21,34,55],
+    ])
+    >>> dism = pairwise_vlr(mat)
+    >>> dism.redundant_form()
+    array([[0.        , 0.01576411, 0.00649553],
+           [0.01576411, 0.        , 0.00202147],
+           [0.00649553, 0.00202147, 0.        ]])
+
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Mask zeros
+    mat = mat.astype(np.float64)
+    
```

Comment:
```suggestion
    mat = closure(mat)
```
This will ward against the user inputting negative proportions